src/libraries/Common/src/System/IO/FileSystem.DirectoryCreation.Windows.cs

```diff
@@ -34,39 +34,49 @@ public static unsafe void CreateDirectory(string fullPath, byte[]? securityDescr
             // isn't threadsafe.
 
             bool somepathexists = false;
-            int length = fullPath.Length;
+
+            ReadOnlySpan<char> fullPathSpan = fullPath;
+            int length = fullPathSpan.Length;
 
             // We need to trim the trailing slash or the code will try to create 2 directories of the same name.
-            if (length >= 2 && PathInternal.EndsInDirectorySeparator(fullPath.AsSpan()))
+            if (length >= 2 && PathInternal.IsDirectorySeparator(fullPathSpan[^1]))
```

Comment:
does it save anything to avoid calling AsSpan() twice? I would have thought that is very fast.

Reply:
I actually assumed that this would change something. 
But this intrinsic is probably optimized well enough.

Of course, I find the introduction of a local more readable. 

A good rule of thumb is, if you know the output of a function is deterministic and you use that output multiple times, make a local output for it!

@danmoseley 
In that case, however, we have no performance improvement from it (CodeGen is the same).

Should I remove this?

Reply:
@danmoseley friendly ping :smile:

Reply:
oops. Yes, I'd keep it as was. AsSpan() is idiomatic - we use it in the hottest places.